src/backend/base/langflow/base/agents/agent.py

```diff
@@ -178,6 +179,18 @@ async def run_agent(
     def create_agent_runnable(self) -> Runnable:
         """Create the agent."""
 
+    def validate_tool_names(self) -> None:
+        """Validate tool names to ensure they match the required pattern."""
+        pattern = re.compile(r"^[a-zA-Z0-9_-]+$")
+        if hasattr(self, "tools") and self.tools:
+            for tool in self.tools:
+                if not pattern.match(tool.name):
+                    msg = (
+                        f"Invalid tool name '{tool.name}': must only contain letters, numbers, underscores, dashes,"
```

Comment:
what do you think about just taking out the spaces instead of throwing an error?

Reply:
Actually, I thought of that too. Let’s discuss it with Simon as well.

There was a similar scenario in Supabase, where they renamed a collection to have capital letters, and it ended up creating a poor user experience. We also received some queries related to this issue in our Supabase component.

Keeping this in mind, I believe it would be beneficial to discuss it with the team. Does that sound good?